ts/store/reducers/entities/services/servicesByOrganizationFiscalCode.ts

```diff
@@ -55,28 +57,31 @@ export function serviceIdsByOrganizationFiscalCodeReducer(
       const serviceTuples = action.payload;
 
       // Remove service id from the array keyed by organizationFiscalCode
-      const stateUpdate = serviceTuples.reduce<
-        ServiceIdsByOrganizationFiscalCodeState
-      >((accumulator, tuple) => {
-        const serviceId = tuple.e1;
-        const organizationFiscalCode = tuple.e2;
-        const ids = fromNullable(organizationFiscalCode)
-          .map(_ => (accumulator[_] !== undefined ? accumulator[_] : state[_]))
-          .toNullable();
-        if (organizationFiscalCode && ids) {
-          const filteredIds = ids.filter(id => id !== serviceId);
-          return {
-            ...accumulator,
-            [organizationFiscalCode]: filteredIds
-          };
-        }
-        return accumulator;
-      }, {});
-
-      return {
-        ...state,
-        ...stateUpdate
-      };
+      return serviceTuples.reduce<ServiceIdsByOrganizationFiscalCodeState>(
+        (accumulator, tuple) => {
+          const serviceId = tuple.e1;
+          const organizationFiscalCode = tuple.e2;
+          const ids = fromNullable(organizationFiscalCode)
+            .map(
+              _ => (accumulator[_] !== undefined ? accumulator[_] : state[_])
+            )
+            .toNullable();
+          if (organizationFiscalCode && ids) {
+            const filteredIds = ids.filter(id => id !== serviceId);
+            const result = {
+              ...accumulator,
+              [organizationFiscalCode]: filteredIds
+            };
+            if (!filteredIds.length) {
+              // tslint:disable-next-line no-object-mutation
+              delete result[organizationFiscalCode];
+            }
+            return result;
+          }
+          return accumulator;
```

Comment:
```suggestion
          // If any other service is related to the organization, remove the related item from the state
            // otherwise, remove the service id from the item
            return !filteredIds.length ?
              accumulator : {
                ...accumulator,
                [organizationFiscalCode]: filteredIds
              };
          } else return accumulator;
```

Reply:
You can avoid the usage of `delete` thanks to the reduce itself

Reply:
this modification is not sufficient because the "reduce" is initialized with the current state, therefore there is the need to eliminate the data that are no longer needed.

If you want to use the "reduce" you have to consider another solution

Reply:
What if you start from `{}` instead of `state`?

Reply:
@valentinacala in this case I should change the logic and no longer use the "reduce" of the only elements to be eliminated, this could worsen performance.